src/signing/signing.service.ts

```diff
@@ -101,3 +104,41 @@ export class VaultSigningService extends SigningService {
     this.logger.log(`Key "${handle}" with address "${address}" was loaded`);
   }
 }
+
+export class FireblocksSigningService extends SigningService {
+  constructor(
+    protected readonly signingManager: FireblocksSigningManager,
+    protected readonly polymeshService: PolymeshService,
+    private readonly logger: PolymeshLogger
+  ) {
+    super();
+    this.logger.setContext(FireblocksSigningService.name);
+  }
+
+  public async getAddressByHandle(handle: string): Promise<string> {
+    const derivePath = this.handleToDerivationPath(handle);
+
+    const key = await this.signingManager.deriveAccount(derivePath);
+    return key.address;
+  }
+
+  private handleToDerivationPath(handle: string): DerivationPath {
+    const sections = handle.split('-').map(Number);
+
+    if (sections.some(isNaN) || sections.length > 3 || handle === '') {
+      throw new AppValidationError(
+        'Fireblocks `signer` field should be 3 integers formatted like: `x-y-z`'
+      );
+    }
+
+    /**
+     * Mainnet should use `595` as the coinType, otherwise it should be `1` to indicate a test net
+     * reference: https://github.com/satoshilabs/slips/blob/2a2f4c79508749f7e679a127d5a56da079b8d2d8/slip-0044.md?plain=1#L32
+     */
+    const coinType = this.signingManager.ss58Format === 12 ? 595 : 1;
+
+    const [accountId, change, accountIndex] = sections;
+
+    return [44, coinType, accountId, change || 0, accountIndex || 0];
+  }
+}
```

Comment:
IMO this should be in its own file

Reply:
I can refactor each manager into its own file.

I was thinking of splitting them up myself, but figured I'd stick with the pattern.